pkg/osquery/extension.go

```diff
@@ -92,6 +92,9 @@ type ExtensionOpts struct {
 	// RunDifferentialQueriesImmediately allows the client to execute a new query the first time it sees it,
 	// bypassing the scheduler.
 	RunDifferentialQueriesImmediately bool
+	// skipHardwareKeysSetup is a flag to indicate if we should skip setting up hardware keys.
+	// This is useful for testing environments where we don't have required hardware.
+	skipHardwareKeysSetup bool
```

Comment:
I like that this is explicit, but it does require changing a lot of test filed. Do you think it makes more sense than switching off the environment?

Reply:
Ehh, I waffled on this one. It felt worse to have logic in the production code to try and figure out if we were under test.